uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/service/UniProtEntryService.java

```diff
@@ -135,6 +139,36 @@ public UniProtKBEntry findByUniqueId(String accession, String fields) {
         }
     }
 
+    public String findAccessionByProteinId(String proteinId) {
+        try {
+            SolrRequest solrRequest =
+                    SolrRequest.builder()
+                            .query(PROTEIN_ID + ":" + proteinId.toUpperCase() + " AND  " + IS_ISOFORM + ":false")
+                            .sorts(solrSortClause.getSort(null))
+                            .rows(NumberUtils.INTEGER_TWO)
+                            .build();
+            QueryResult<UniProtDocument> queryResult = repository.searchPage(solrRequest, null);
```

Comment:
I believe, you are getting 2 rows as optimisation which is good. But in the below code, you are filtering non-active entry. Is it possible that the first two result has 1 active and 1 inactive but the 3rd (or 3rd onwards) entry might also exist for the same protein id which is active too. Is this case possible?

Reply:
As far as I understood, we can only have one active ID.
There is the Merged or Demerged use case that we will have the same ID as inactive as well. (this is the 2 entry use case). This is why we filter only active one. For us, the active one is important. Does it make sense?